python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -1407,3 +1477,496 @@ def _compute_model_params(self, wcsf):
+            [inputs["isolatedStarCatalogs"][t] for t in inputISSTracts.argsort()]
+        )
+
+        sampleRefCat = inputs["referenceCatalog"][0].get()
+        refEpoch = sampleRefCat[0]["epoch"]
+
+        refConfig = LoadReferenceObjectsConfig()
+        refConfig.anyFilterMapsToThis = "phot_g_mean"
+        refConfig.requireProperMotion = True
+        refObjectLoader = ReferenceObjectLoader(
+            dataIds=[ref.datasetRef.dataId for ref in inputRefCatRefs],
+            refCats=inputs.pop("referenceCatalog"),
+            config=refConfig,
+            log=self.log,
+        )
+
+        output = self.run(
+            **inputs, instrumentName=instrumentName, refEpoch=refEpoch, refObjectLoader=refObjectLoader
+        )
+
+        for outputRef in outputRefs.outputWcs:
+            visit = outputRef.dataId["visit"]
+            butlerQC.put(output.outputWCSs[visit], outputRef)
+        butlerQC.put(output.outputCatalog, outputRefs.outputCatalog)
+        butlerQC.put(output.starCatalog, outputRefs.starCatalog)
+
+    def run(
+        self,
+        inputVisitSummaries,
+        isolatedStarSources,
+        isolatedStarCatalogs,
+        instrumentName="",
+        refEpoch=None,
+        refObjectLoader=None,
+    ):
+        """Run the WCS fit for a given set of visits
+
+        Parameters
+        ----------
+        inputVisitSummaries : `list` of `lsst.afw.table.ExposureCatalog`
+            List of catalogs with per-detector summary information.
+        instrumentName : `str`, optional
+            Name of the instrument used. This is only used for labelling.
+        refEpoch : `float`
+            Epoch of the reference objects in MJD.
+        refObjectLoader : instance of
+            `lsst.meas.algorithms.loadReferenceObjects.ReferenceObjectLoader`
+            Referencef object loader instance.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            ``outputWCSs`` : `list` of `lsst.afw.table.ExposureCatalog`
+                List of exposure catalogs (one per visit) with the WCS for each
+                detector set by the new fitted WCS.
+            ``fitModel`` : `wcsfit.WCSFit`
+                Model-fitting object with final model parameters.
+            ``outputCatalog`` : `pyarrow.Table`
+                Catalog with fit residuals of all sources used.
+        """
+        self.log.info("Gathering instrument, exposure, and field info")
+        # Set up an instrument object
+        instrument = wcsfit.Instrument(instrumentName)
+
+        # Get information about the extent of the input visits
+        fields, fieldRegions = self._prep_sky(inputVisitSummaries)
+
+        # Get RA, Dec, MJD, etc., for the input visits
+        exposureInfo, exposuresHelper, extensionInfo = self._get_exposure_info(
+            inputVisitSummaries, instrument, fieldRegions=fieldRegions
+        )
+
+        self.log.info("Load associated sources")
```

Comment:
I changed the tense above instead to "Gather" rather than "Gathering". They're not really useful unless you're actively monitoring the run. I'm probably going to reevaluate the logs in general on a separate ticket.